enterprise/frontend/src/embedding-sdk/components/private/Breadcrumb/utils/update-breadcrumbs.ts

```diff
@@ -0,0 +1,46 @@
+import type { SdkBreadcrumbItem } from "embedding-sdk/types/breadcrumb";
+
+export function updateBreadcrumbsWithItem(
+  breadcrumbs: SdkBreadcrumbItem[],
+  nextItem: SdkBreadcrumbItem,
+): SdkBreadcrumbItem[] {
+  if (breadcrumbs.length === 0) {
+    return [nextItem];
+  }
+
+  const lastItem = breadcrumbs[breadcrumbs.length - 1];
+
+  // When navigating to a dashboard card, do not show in breadcrumbs
+  // for consistency with ad-hoc questions.
+  if (lastItem.type === "dashboard" && nextItem.type === "question") {
+    return breadcrumbs;
+  }
+
+  if (nextItem.type === "collection") {
+    return updateBreadcrumbsWithCollection(breadcrumbs, nextItem);
+  }
+
+  // If a dashboard or question already exist as the last item, replace it.
+  if (lastItem.type === nextItem.type) {
+    return [...breadcrumbs.slice(0, -1), nextItem];
+  }
+
+  return [...breadcrumbs, nextItem];
+}
```

Comment:
Ideally entity-specific logic should override default breadcrumb `updateItem` logic only for some entities, to make breadcrumbs implementation entity-details-independent:
```
reportLocation({
        type: "dashboard",
        id: dashboard.id,
        name: dashboard.name,
        updateBreadcrumbs: updateBreadcrumbsForDashboard // <== something like this
        onNavigate: onNavigateBackToDashboard,
})
```

but probably for simplicity we can keep it as is at least until more entities are added

Reply:
Nice! I'll keep it as-is for now till we have more entities 🙏🏻